lib/OptionsDefaulter.js

```diff
@@ -27,19 +21,21 @@ function setProperty(obj, name, value) {
 	obj[name.pop()] = value;
 }
 
-OptionsDefaulter.prototype.process = function(options) {
-	var addItemTo = function addItemTo(list) {
-		return function(item) {
-			list.push(item);
-		}
+class OptionsDefaulter {
+	constructor() {
+		this.defaults = {}
+		this.config = {};
 	}
 
-	for(var name in this.defaults) {
-		switch(this.config[name]) {
-			case undefined:
-				if(getProperty(options, name) === undefined)
-					setProperty(options, name, this.defaults[name]);
-				break;
+	process(options) {
+		const addItemTo = (list) => (item) => list.push(item);
```

Comment:
This will make stack trace less readable because of lack of `function.name` in debugger